community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/state/DefaultIndexProviderMap.java

```diff
@@ -96,15 +98,22 @@ public IndexProvider lookup( String providerDescriptorName ) throws IndexProvide
         throw notFound( providerDescriptorName );
     }
 
-    private IllegalArgumentException notFound( Object key )
+    private IllegalArgumentException notFound( String name )
     {
-        return new IllegalArgumentException( "Tried to get index provider for an existing index with provider " + key +
-                " whereas available providers in this session being " + indexProvidersByDescriptor + ", and default being " + defaultIndexProvider );
+        return new IllegalArgumentException( "Tried to get index provider with name " + name +
+                " whereas available providers in this session being " + Arrays.toString( indexProviderNames() ) + ", and default being " +
+                defaultIndexProvider.getProviderDescriptor().name() );
     }
 
     @Override
     public void accept( Consumer<IndexProvider> visitor )
     {
         indexProvidersByDescriptor.values().forEach( visitor );
     }
+
+    private String[] indexProviderNames()
```

Comment:
There's no need in this method, using `indexProvidersByName.keySet()` will render exactly the same string representation.